### ISSUE TITLE
Invalid character escape '\d'. when trying to build on Android [Windows 11]

### ISSUE THREAD

**Issue by vreoo** [software-mansion/react-native-reanimated#4349]
### Description

I updated `react-native-reanimated` from `v1.13.4` to `v2.10.0`.

Using a macOS device the build was successful and no issues have been found. But on **Windows** I have been struggling to make this build successfully.

Once I run `yarn run android` or try to "sync the project with Gradle files" in Android Studio, I get this error:
```
error when building with cmake using C:\development\yuehlia\node_modules\react-native-reanimated\android\CMakeLists.txt: Build command failed.
Error while executing process C:\Users\Yuehl\AppData\Local\Android\Sdk\cmake\3.18.1\bin\cmake.exe with arguments {-HC:\development\yuehlia\node_modules\react-native-reanimated\android -DCMAKE_CXX_FLAGS=-fstack-protector-all -DCMAKE_FIND_ROOT_PATH=C:\development\yuehlia\node_modules\react-native-reanimated\android\.cxx\cmake\debug\prefab\x86\prefab -DCMAKE_BUILD_TYPE=Debug -DCMAKE_TOOLCHAIN_FILE=C:\Users\Yuehl\AppData\Local\Android\Sdk\ndk\21.4.7075529\build\cmake\android.toolchain.cmake -DANDROID_ABI=x86 -DANDROID_NDK=C:\Users\Yuehl\AppData\Local\Android\Sdk\ndk\21.4.7075529 -DANDROID_PLATFORM=android-21 -DCMAKE_ANDROID_ARCH_ABI=x86 -DCMAKE_ANDROID_NDK=C:\Users\Yuehl\AppData\Local\Android\Sdk\ndk\21.4.7075529 -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_LIBRARY_OUTPUT_DIRECTORY=C:\development\yuehlia\node_modules\react-native-reanimated\android\build\intermediates\cmake\debug\obj\x86 -DCMAKE_RUNTIME_OUTPUT_DIRECTORY=C:\development\yuehlia\node_modules\react-native-reanimated\android\build\intermediates\cmake\debug\obj\x86 -DCMAKE_MAKE_PROGRAM=C:\Users\Yuehl\AppData\Local\Android\Sdk\cmake\3.18.1\bin\ninja.exe -DCMAKE_SYSTEM_NAME=Android -DCMAKE_SYSTEM_VERSION=21 -DANDROID_STL=c++_shared -DNATIVE_DEBUG=false -DREACT_NATIVE_TARGET_VERSION=65 -DANDROID_TOOLCHAIN=clang -DBOOST_VERSION=1_63_0 -DNODE_MODULES_DIR=C:\development\yuehlia\node_modules -DJS_RUNTIME=jsc -DJS_RUNTIME_DIR=C:\development\yuehlia\node_modules\react-native\ReactCommon\jsi -DCLIENT_SIDE_BUILD=true -DIS_NEW_ARCHITECTURE_ENABLED=false -DPLAYGROUND_APP_NAME=NOT_FOUND -DRNVERSION=65 -BC:\development\yuehlia\node_modules\react-native-reanimated\android\.cxx\cmake\debug\x86 -GNinja}
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Users/Yuehl/AppData/Local/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/windows-x86_64/bin/clang.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Users/Yuehl/AppData/Local/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/windows-x86_64/bin/clang++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring incomplete, errors occurred!
See also "C:/development/yuehlia/node_modules/react-native-reanimated/android/.cxx/cmake/debug/x86/CMakeFiles/CMakeOutput.log".

CMake Error at CMakeLists.txt:74 (add_library):
  Syntax error in cmake code when parsing string

    C:\development\yuehlia\node_modules/react-native/ReactCommon/jsi/jsi/jsi.cpp

  Invalid character escape '\d'.


CMake Error at CMakeLists.txt:74 (add_library):
  Syntax error in cmake code when parsing string

    C:\development\yuehlia\node_modules/react-native/ReactCommon/jsi/jsi/JSIDynamic.cpp

  Invalid character escape '\d'.
```

I tried to fix it so many times (Updating Cmake / use backslash / etc..)

I've searched the repo for a solution to this, but it seems that there is no issue raised for this particular error.

### Steps to reproduce

1. Update `react-native-reanimated` from version `v1.13.4` to `v2.10.0`
2. Run `yarn cache clean --all && yarn install && cd android && ./gradlew clean && ./gradlew cleanBuildCache && cd ..`
3. Run `yarn run android`
4. I get the error

Android Studio:
1. Update `react-native-reanimated` from version `v1.13.4` to `v2.10.0`
2. Sync the project with Gradle files.
3. I get the error

### Snack or a link to a repository

not needed

### Reanimated version

2.10.0

### React Native version

0.65.3

### Platforms

Android, iOS

### JavaScript runtime

JSC

### Workflow

React Native (without Expo)

### Architecture

Paper (Old Architecture)

### Build type

Release mode

### Device

Real device

### Device model

Huawei nova 7i - JNY-LX1

### Acknowledgements

Yes

**Comment by vreoo** [comment]
Apparently, there is a fix for this issue introduced in [v2.13.0](https://github.com/software-mansion/react-native-reanimated/releases/tag/2.13.0). But when building with this version of `react-native-reanimated` It builds successfully but when trying to run it on Android it fails to start the application.

I checked the CI of this PR #3755, But it seems it also fails to build Android for `react-native` v0.65.

I am not sure what to do from here.

**Comment by tomekzaw** [comment]
Hey @vreoo, thanks for reporting this issue. Indeed, the problem with Windows paths was fixed in #3755 and released in 2.13.0. Please note that the latest available version of Reanimated v2 is 2.15.0. If you don't use Reanimated v1 API, you should upgrade to Reanimated v3 (latest is 3.0.2), for details see [this blog post](https://blog.swmansion.com/releasing-reanimated-3-0-17fab4cb2394).

As for the error, have you actually tried doing the recommended steps from the error message?
1. Make sure `react-native-reanimated/plugin` is listed in your `babel.config.js`, see installation docs at https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/installation#babel-plugin
2. Make sure you reset build cache after updating the config, run: `yarn start --reset-cache`

**Comment by vreoo** [comment]
Hello @tomekzaw, Thank you for replying back to me.

> 1. Make sure `react-native-reanimated/plugin` is listed in your `babel.config.js`, see installation docs at https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/installation#babel-plugin
> 2. Make sure you reset build cache after updating the config, run: `yarn start --reset-cache`

These steps you provided fixed my issue, I cleared all the caches (yarn & Android build) and ran `yarn install`, and build on Android. All is working fine and the app is running on my machine.

Thank you for your help.
